### PR TITLE
Add DrawPointsStep and DrawHeatmapStep

### PR DESCRIPTION
**Pull request by treo**
DrawPointsStep takes a point or a list of points and a mapping for their labels to colors, and draws those on a black background. 

DrawHeatmapStep takes a point or a list of points and draws a heatmap. It collects point positions over time, so depending on the fading factor, old "heat" is retained.

Both steps pass through all data that is given to them - I'm not sure if that is something that they should do, as not all steps actually do that. On some it is a configuration option, others just don't do it.

And another question on style: As we have lombok in the project, we can use `val`, so far I've tried to stay as close to plain java as possible. What is the preferred style there?

### CODE REVIEW

File: konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStepRunner.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            width = step.width();
+            height = step.height();
+        }else{
+            throw new IllegalArgumentException("You have to provide either a reference image or width AND height!");
+        }
+
+        // turn points with relative addressing to absolute addressing
+        List<Point> absPoints = new ArrayList<>(points.size());
+        for (Point point : points) {
+            absPoints.add(point.toAbsolute(width, height));
```
Comment (AlexDBlack) [inline konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStepRunner.java:115]: We should probably mention that it accepts both types of points in the doc

File: konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStepRunner.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        }
+
+        // create empty image
+        Mat image = new Mat();
+        image.put(Mat.zeros(height, width, CvType.CV_8UC3));
```
Comment (AlexDBlack) [inline konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStepRunner.java:120]: As mentioned earlier - let's allow drawing on the image as an option too

File: konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/heatmap/DrawHeatmapStep.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ *
+ * @author Paul Dubs
+ */
+@Builder
+@Data
+@Accessors(fluent = true)
+@AllArgsConstructor
+@JsonName("DRAW_HEATMAP")
+@Schema(description = "A pipeline step that configures how to draw 2D points, optionally on an image.")
+public class DrawHeatmapStep implements PipelineStep {
```
Comment (AlexDBlack) [inline konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/heatmap/DrawHeatmapStep.java:54]: Again, "on image" vs. "image as size reference".
Being able to overlay on an image would definitely be a nice addition (at the very least, we'll open an issue and add it later)

File: konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/heatmap/DrawHeatmapStep.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    public static final String DEFAULT_OUTPUT_NAME = "image";
+
+    @Schema(description = "Names of the points to be used for the heatmap. Accepts both single points and lists of points.")
+    @Singular
+    private List<String> points;
```
Comment (AlexDBlack) [inline konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/heatmap/DrawHeatmapStep.java:59]: Same docs clarification issue

File: konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/bbox/point/BoundingBoxToPointStep.java [hunk 5]
```diff
@@ -35,6 +35,7 @@
  * the output will be a single value; if the input is a list, the output will be a list.<br>
  */
 @Data
+@Builder
```
Comment (AlexDBlack) [inline konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/bbox/point/BoundingBoxToPointStep.java:38]: No builder: we're standardizing on fluent API everywhere - https://github.com/KonduitAI/konduit-serving/issues/390

File: konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStep.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+ * </ul>
+ *
+ * @author Paul Dubs
+ */
+@Builder
+@Data
+@Accessors(fluent = true)
+@AllArgsConstructor
+@JsonName("DRAW_POINTS")
+@Schema(description = "A pipeline step that configures how to draw 2D points, optionally on an image.")
```
[the remark this answers is not in the document]
Comment (treo) [reply konduit-serving-data/konduit-serving-image/src/main/java/ai/konduit/serving/data/image/step/point/draw/DrawPointsStep.java:56]: Ah, forgot to change it here.

I initially wanted to draw on top of the image, but decided against it, the main reason being that image composition should probably be a step of its own.

**Pull request merged**